### DISCUSSION TITLE
It is posible to test DGS Mutations without code gen?

### DISCUSSION THREAD

**Discussion by sbarroso-RatedPower** [Netflix/dgs-framework#1072] [Q&A] [answered]
Hi everyone,

I'm trying to test some DGS mutations without using the code gen. There is a way to do it?

That's the example with the way that I'm trying to test it:

```java
    @Test
    void whenQueryRoleById_ThenReturnsCorrectly() {

        when(roleService.create(createRoleDTO)).thenReturn(mockRoleDTO());

        String resultId = dgsQueryExecutor.executeAndExtractJsonPath(
                "{ addFoo (foo: 0) { id }}",
                "data.foo.id"
        );

        assertNotNull(resultId);
    }
```

Thanks in advance!

**Comment by srinivasankavitha** [comment]
Codegen is optional. You can use any POJO that maps to your schema type. So yes, you can test without codegen classes.

**Reply by sbarroso-RatedPower** [reply]
Hi, thank for your answer. Do you know where I can find an example of that ?

Or How Can I use a POJO to map my schema type ?

Thank you in adcance

**Comment by hantsy** [comment, marked as answer]
Check my example, https://github.com/hantsy/spring-graphql-sample/tree/master/dgs which create your POJOs maps to graphql schemas, https://github.com/hantsy/spring-graphql-sample/tree/master/dgs/src/main/java/com/example/demo/gql/types

**Reply by sbarroso-RatedPower** [reply]
Thank you !

**Comment by srinivasankavitha** [comment]
So if you already have a defined class, you can configure codegen (assuming
you need to use codegen for some types?) by adding a typemapping:
https://netflix.github.io/dgs/generating-code-from-schema/#mapping-existing-types

If you are not using codegen types at all, you just use the class that you
have defined for that schema.

Hope that helps.
